openedx/core/lib/block_structure/transformer_registry.py

```diff
@@ -31,6 +35,22 @@ def get_registered_transformers(cls):
             return set()
 
     @classmethod
+    @memoized
+    def get_write_version_hash(cls):
+        """
+        Returns a deterministic hash value of the WRITE_VERSION of all
+        registered transformers.
+        """
+        hash_obj = sha1()
+
+        sorted_transformers = sorted(cls.get_registered_transformers(), key=lambda t: t.name())
+        for transformer in sorted_transformers:
+            hash_obj.update(transformer.name())
+            hash_obj.update(unicode(transformer.WRITE_VERSION))
```

Comment:
Sounds good.  I had considered issue 1 but deemed it unnecessary for the same reason.  
But hadn't known about issue 2.  Glad to have a Python-3 expert on the team.